packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -100,119 +124,132 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
     initialize(): void {
         this.initializeProviders();
     }
-    protected async initializeProviders(): Promise<void> {
+
+    protected initializeProviders(): void {
         try {
-            const providers = this.createProviders();
-            this.toDispose.pushAll(providers);
-            await Promise.all(providers.map(p => p.ready));
+            this.createProviders();
             if (this.toDispose.disposed) {
                 return;
             }
-            this.providers.push(...providers);
-            for (const provider of providers) {
-                provider.onDidPreferencesChanged(_ => this.reconcilePreferences());
+            for (const [scope, provider] of this.providersMap.entries()) {
+                if (scope !== PreferenceScope.DEFAULT) {
+                    this.toDispose.push(provider.onDidPreferencesChanged(changes =>
+                        this.reconcilePreferences(changes)
+                    ));
+                }
             }
-            this.reconcilePreferences();
-            this._ready.resolve();
+            Promise.all(this.providers.map(p => p.ready)).then(() => this._ready.resolve());
         } catch (e) {
             this._ready.reject(e);
         }
     }
-    protected createProviders(): PreferenceProvider[] {
-        return [
-            this.providerProvider(PreferenceScope.User),
-            this.providerProvider(PreferenceScope.Workspace)
-        ];
-    }
-
-    protected reconcilePreferences(): void {
-        const changes: PreferenceChanges = {};
-        const deleted = new Set(Object.keys(this.preferences));
-        const preferences = this.parsePreferences();
-        // tslint:disable-next-line:forin
-        for (const preferenceName in preferences) {
-            deleted.delete(preferenceName);
-            const oldValue = this.preferences[preferenceName];
-            const newValue = preferences[preferenceName];
-            if (oldValue !== undefined) {
-                if (!JSONExt.deepEqual(oldValue, newValue)) {
-                    changes[preferenceName] = { preferenceName, newValue, oldValue };
-                    this.preferences[preferenceName] = deepFreeze(newValue);
+
+    protected createProviders(): void {
+        PreferenceScope.getScopes().forEach(scope =>
+            this.doCreateProvider(scope)
+        );
+    }
+
+    protected reconcilePreferences(changes: PreferenceChanges): void {
```

Comment:
it looks suspicious now.

Before all scopes were taken into the account, for example, if preferences changed on user level, but it is already overriden on workspace then there should not be any events.
Now it seems to care about only one scope. It is bogus, should be like before.

Reply:
it cares about all scopes, the logic is between lines 162-174